### ISSUE TITLE
Consistency check failed: File has not had execution verified

### ISSUE THREAD

**Issue by echesakovMSFT** [dotnet/coreclr#19668]
The three following assemblies when PMI-d on Linux/arm hit the same assertion

**System.Security.Cryptography.Encoding.dll** 

```
ubuntu@tegra-ubuntu:~$ $CORE_ROOT/corerun /mnt/echesakov/jitutils/bin/pmi.dll PREPALL-QUIET $CORE_ROOT/System.Security.Cryptography.Encoding.dll

Assert failure(PID 6645 [0x000019f5], Thread: 6645 [0x19f5]): Consistency check failed: File has not had execution verifiedFAILED: m_bDisableActivationCheck || CheckLoadLevel(FILE_ACTIVE)
         FAILED: pDomainFile->CheckActivated()
                /opt/code/src/vm/ceeload.cpp, line: 13830
         FAILED: GetModule()->CheckActivated()
                /opt/code/src/vm/methodtable.cpp, line: 9358
         FAILED: CheckActivated()
                /opt/code/src/vm/methodtable.cpp, line: 3699
    File: /opt/code/src/vm/domainfile.cpp Line: 390
    Image: /mnt/echesakov/PR19619/bin/tests/Linux.arm.Checked/Tests/Core_Root/corerun

Aborted
```

**System.Security.Cryptography.OpenSsl.dll**
```
ubuntu@tegra-ubuntu:~$ $CORE_ROOT/corerun /mnt/echesakov/jitutils/bin/pmi.dll PREPALL-QUIET $CORE_ROOT/System.Security.Cryptography.OpenSsl.dll

Assert failure(PID 7778 [0x00001e62], Thread: 7778 [0x1e62]): Consistency check failed: File has not had execution verifiedFAILED: m_bDisableActivationCheck || CheckLoadLevel(FILE_ACTIVE)
         FAILED: pDomainFile->CheckActivated()
                /opt/code/src/vm/ceeload.cpp, line: 13830
         FAILED: GetModule()->CheckActivated()
                /opt/code/src/vm/methodtable.cpp, line: 9358
         FAILED: CheckActivated()
                /opt/code/src/vm/methodtable.cpp, line: 3699
    File: /opt/code/src/vm/domainfile.cpp Line: 390
    Image: /mnt/echesakov/PR19619/bin/tests/Linux.arm.Checked/Tests/Core_Root/corerun

Aborted
```

**System.Security.Cryptography.Algorithms.dll** (reproducible under COMPlus_TieredCompilation=0 only)

```
ubuntu@tegra-ubuntu:~$ COMPlus_TieredCompilation=0 $CORE_ROOT/corerun /mnt/echesakov/jitutils/bin/pmi.dll PREPALL-QUIET $CORE_ROOT/System.Security.Cryptography.Algorithms.dll

Assert failure(PID 9067 [0x0000236b], Thread: 9067 [0x236b]): Consistency check failed: File has not had execution verifiedFAILED: m_bDisableActivationCheck || CheckLoadLevel(FILE_ACTIVE)
         FAILED: pDomainFile->CheckActivated()
                /opt/code/src/vm/ceeload.cpp, line: 13830
         FAILED: GetModule()->CheckActivated()
                /opt/code/src/vm/methodtable.cpp, line: 9358
         FAILED: CheckActivated()
                /opt/code/src/vm/methodtable.cpp, line: 3699
    File: /opt/code/src/vm/domainfile.cpp Line: 390
    Image: /mnt/echesakov/PR19619/bin/tests/Linux.arm.Checked/Tests/Core_Root/corerun

Aborted
```
To reproduce - build https://github.com/dotnet/jitutils first and then run the command 

`$CORE_ROOT/corerun $JitUtils/bin/pmi.dll PREPALL-QUIET $CORE_ROOT/System.Security.Cryptography.OpenSsl.dll`

**Updated:** Also reproduces on Linux/arm64

**Comment by AndyAyersMS** [comment]
I'll investigate.

**Comment by jkotas** [comment]
> Wonder if there is some activation check missing from `PrepareMethod`

Yes, looks like PrepareMethod is missing `pMD->EnsureActive()`

**Comment by AndyAyersMS** [comment]
Fixed by #22351.
